dist/api_err.py

```diff
@@ -51,6 +51,11 @@ def __init__(self, name, value, desc, long_desc=None, **flags):
         'recovery must be run to continue', '''
         This error is generated when wiredtiger_open is configured
         to return an error if recovery is required to use the database.'''),
+    Error('WT_CACHE_FULL', -31807,
+        'operation would overflow cache', '''
+        This error is generated when wiredtiger_open is configured
+        to run in-memory, and an insert or update operation requires more
+        than the configured cache size to complete.'''),
```

Comment:
I'm sympathetic to both points of view, but the setting that configures the amount of memory available is going to remain "cache_size", and all of the relevant statistics will be "cache" statistics.  I don't think we can hide that from in-memory users without major surgery.

How about we make this `undoc=True` for now, since the whole feature is currently undocumented: that buys us some time to figure out how we're going to deal with this in MongoDB?